scene/2d/animated_sprite_2d.cpp

```diff
@@ -154,29 +161,22 @@ void AnimatedSprite2D::_validate_property(PropertyInfo &p_property) const {
 void AnimatedSprite2D::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_INTERNAL_PROCESS: {
-			if (frames.is_null()) {
+			if (frames.is_null() || !frames->has_animation(animation)) {
 				return;
 			}
-			if (!frames->has_animation(animation)) {
-				return;
-			}
-			if (frame < 0) {
-				return;
+
+			double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
+			if (speed == 0) {
```

Comment:
Shouldn't it use `is_zero_approx()`?

Reply:
Yeah, it was like that. It makes sense to me, at least. Even if outright improbable that someone would find themselves using such a low value, you'd want to account for even the smallest speed to make sure the frames progress correctly.